### PR TITLE
OpenShift Reference Component Docs

### PR DESCRIPTION
**Pull request by mtnbikenc**
Provide documentation on how to add support for components in OpenShift-Ansible

https://trello.com/c/z1MuGbJe/634-5-cloud-native-infra-reference-ansible-role-for-teams-to-follow

### CODE REVIEW

**Comment by michaelgugino** [comment]
Cleanup PR: https://github.com/openshift/openshift-ansible/pull/7256

/lgtm

**Comment by mtnbikenc** [comment]
@kwoodson I'd like your feedback on usage of `oc_` modules and if any of that needs to be cleaned up or improved so we can better document those steps.

**Comment by kwoodson** [comment]
@mtnbikenc,  great work here.

I think the single item that we cannot stress enough is idempotency.  This is often overlooked or under emphasized as people interpret rerunnable as equivalent to idempotency.  

'''
an idempotent operation is one that has no additional effect if it is called more than once with the same input parameters
'''

@mtnbikenc, the comment regarding the `oc_modules` or `lib_openshift` is a good one.  In the near future, we might be moving away from those modules so I can't put a hard requirement on their usage.  What I would be clear about in this document is to maybe have a few rules or policies:

Prioritize:  ( we can discuss this order )
1. idempotence is most important
2. maintainability
3. modularity is second

When is it better to use ansible?
- **do** use modules provided by ansible
-  **limit** use of command and shell modules unless there isn't another module or the task can be accomplished with ansible

When to write custom modules?
- when logic gets too complex 
  - lots of if statements
  - complex loops
  - complex variables or data structures
- refrain from writing 20-30+ tasks in a single task/playbook
  - this generally means the logic is too complex or isn't single purpose enough
- when error handling can be improved
- library-like: when multiple users could benefit from the reuse of the functionality

General advice:
- try not to leave artifacts like files, directories, etc
- **avoid** failed_when whenever possible
- always name your tasks!
- document your code
- follow good role naming scheme (team to decide or finalize)
- use inventory variables whenever possible with sane defaults
- avoid set_facts

I know this is a bit to ask but this should be our goal.  This is going to be the key to achieving success.

@michaelgugino @mtnbikenc @sdodson @vrutkovs 
thoughts?

File: playbooks/openshift-component_name/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# OpenShift Component_Name Playbooks
```
Comment (chrrrles) [inline playbooks/openshift-component_name/README.md]: Are these playbook and role files to be included with the final PR?

Comment (mtnbikenc) [reply playbooks/openshift-component_name/README.md]: No they are not.  I'll pull them before I take the WIP off.

**Comment by mtnbikenc** [comment]
I was hung up on trying to also update the best_practices_guide and the core_concepts_guide in this effort.  I've reverted to just including the relevant details in this guide and we can work to update the other guides in a Trello card for a future release.

**Pull request merged**